nix/dev-setup.nix

```diff
@@ -0,0 +1,43 @@
+# A module to ease setting up a local PostgreSQL database to start with development.
+# This lets you specify your local unix user as which you run the tracker and
+# gives it full access to the nix-security-tracker database user.
+
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.nix-security-tracker-dev-environment;
+in
+{
+
+  options.nix-security-tracker-dev-environment = {
+    enable = lib.mkEnableOption (lib.mdDoc "development environment for nix-security-tracker");
+    user = lib.mkOption {
+      type = lib.types.str;
+      description = "Unix user that runs the nix-security-tracker to connect to the database";
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    services.postgresql = {
+      enable = true;
+      ensureDatabases = [ "nix-security-tracker" ];
+      ensureUsers = [
+        {
+          name = "nix-security-tracker";
+          ensureDBOwnership = true;
+        }
+      ];
+      extraPlugins = with config.services.postgresql.package.pkgs; [ postgis ];
```

Comment:
Hm, do we need PostGIS for something?

Reply:
Oh my bad, that was copied from something else!